### ISSUE TITLE
Outputs from Multi-Output nodes not always passed correctly

### ISSUE THREAD

**Issue by klucknav** [AcademySoftwareFoundation/MaterialX#835]
It seems that when a multi-output node is directly providing the nodegraph output, only one of the outputs is actually used. There doesn't seem to be any issue if the outputs are passed through another node.

To see this in the MaterialXViewer, below is a test mtlx file with an ```<artistic_ior>``` node in the nodegraph. When the ```<artistic_ior>``` outputs are passed through a no-opt ```<dot>``` node, the sphere is the expected red color. However, when the nodegraph outputs are coming directly from the ```<artistic_ior>``` node, the sphere renders as a cyan color. 

The underlying issue can be seen in the main function of the generated Pixel Shader. The two different outputs are not being appropriately fed into ```IMPL_standard_surface_surfaceshader()```. Only the first output (ior in this case) is being used, and is passed in instead of the second output.

```XML
<?xml version="1.0"?>
<materialx version="1.38">

  <nodegraph name="test_ng">
    <artistic_ior name="artistic_ior" type="multioutput">
      <input name="reflectivity" type="color3" value="1,0,0" />
      <input name="edge_color" type="color3" value="1,1,1" />
    </artistic_ior>

    <!-- Option 1 - the expected red color -->
    <dot name="specular" type="color3">
      <input name="in" type="color3" nodename="artistic_ior" output="extinction" />
    </dot>
    <dot name="ior" type="color3">
      <input name="in" type="color3" nodename="artistic_ior" output="ior" />
    </dot>
    <output name="base_color_output" type="color3" nodename="specular"/>
    <output name="specular_color_output" type="color3" nodename="ior"/>

    <!-- Option 2 - a cyan color -->
    <!-- <output name="base_color_output" type="color3" nodename="artistic_ior" output="extinction" />
    <output name="specular_color_output" type="color3" nodename="artistic_ior" output="ior" /> -->
  </nodegraph>

  <standard_surface name="test_sr" type="surfaceshader">
    <input name="base_color" type="color3" output="base_color_output" nodegraph="test_ng" />
    <input name="specular_color" type="color3" output="specular_color_output" nodegraph="test_ng" />
  </standard_surface>
  <surfacematerial name="test_m" type="material">
    <input name="surfaceshader" type="surfaceshader" nodename="test_sr" />
  </surfacematerial>
</materialx>
```

**Comment by kwokcb** [comment]
This is another example case found by Rasmus Bonnedal:
```xml
<?xml version="1.0"?>
<materialx version="1.38">
  <nodegraph name="NG_untitled">
    <separate3 name="node2" type="multioutput" nodedef="ND_separate3_color3">
      <input name="in" type="color3" value="1.0, 0.0, 0.0" />
    </separate3>
    <output name="output2" type="float" nodename="node2" output="outg" />
  </nodegraph>
  <standard_surface name="SR_untitled" type="surfaceshader">
    <input name="base" type="float" output="output2" nodegraph="NG_untitled" />
  </standard_surface>
  <surfacematerial name="Untitled" type="material">
    <input name="surfaceshader" type="surfaceshader" nodename="SR_untitled" />
  </surfacematerial>
</materialx>
```

**Comment by jstone-lucasfilm** [comment]
Thanks to @klucknav for originally posting this report, and thanks to @kwokcb for the fix!  Pull request #853 has been merged to main, and should be part of the upcoming 1.38.4 release.
